internal/fields/sql.go

```diff
@@ -82,6 +94,12 @@ func (f Valuer) Value() (driver.Value, error) {
 		if iface, ok := i.(encoding.BinaryMarshaler); ok {
 			return iface.MarshalBinary()
 		}
+		if v, ok := nonPointerProtoMessage(f.Descriptor, f.value); ok {
```

Comment:
when do we end up with a non-pointer? i'd expect proto structs to be big and probably don't want to copy them

Reply:
It's the non-pointer case:
```
type MyType struct {
  MyProto proto.SomeProtoType
}
```
As opposed to:
```
type MyType struct {
  MyProto *proto.SomeProtoType
}
```